shared/chat/manage-channels/edit-channel-container.js

```diff
@@ -14,95 +15,79 @@ const mapStateToProps = (state: TypedState, {navigateUp, routePath, routeProps})
     throw new Error('conversationIDKey unexpectedly empty')
   }
 
-  const teamname = routeProps.get('teamname')
-  if (!teamname) {
-    throw new Error('teamname unexpectedly empty')
+  // We could try to load the particular channel here, but we should
+  // already have the meta in the store.
+  const meta = maybeGetMeta(state, conversationIDKey)
+  if (!meta) {
+    throw new Error(`meta unexpectedly empty for ${conversationIDKey}`)
   }
 
-  const channelInfo = Constants.getChannelInfoFromConvID(state, conversationIDKey)
-  const _needsLoad = !channelInfo
+  const teamname = meta.teamname
 
-  const waitingForGetInfo = _needsLoad || anyWaiting(state, Constants.getChannelsWaitingKey(teamname))
-  const waitingForUpdate = anyWaiting(
+  const waitingForSave = anyWaiting(
```

Comment:
Not sure if we even still want to have these waiting keys, since we close out the dialog on save. I guess this handles the case where you save and then immediately go back into the dialog, but that seems like an edge case. Thoughts?